src/chainstate/stacks/block.rs

```diff
@@ -162,6 +162,8 @@ impl StacksBlockHeader {
     }
 
     pub fn is_genesis(&self) -> bool {
+        // Aaron: not sure about this -- wouldn't the FIRST_STACKS_BLOCK_HASH _be_ the genesis block?
+        //   a child of that block wouldn't be the genesis block.
```

Comment:
I agree that the naming could be better.  The first _mined_ block has its parent block hash set to all 0's (`FIRST_STACKS_BLOCK_HASH`).  The first _ever_ block in the system, however, is the block that contains the boot code.  You can see this in `install_boot_code()`, where we calculate the index block hash of the boot code block from `FIRST_STACKS_BLOCK_HASH` and `FIRST_BURNCHAIN_BLOCK_HASH`.  You can also see the burndb entry for it under `FIRST_BURNCHAIN_BLOCK_HASH`.

Reply:
Right -- I think the issue was that this caused `StacksChainState::get_index_hash` for the first _mined_ block (whose parent is all 0's) to return the block sentinel, which should not be the index hash for the first mined block.

Reply:
Changed this comment to a note that this will return true for the first mined block.